Resources/public/js/apps/ez-platformuiapp.js

```diff
@@ -140,8 +142,20 @@ YUI.add('ez-platformuiapp', function (Y) {
 
             this.on({
                 'loadingChange': this._loading,
-                'navigate': function () {
+                '*:contentDiscover': function() { this.set('isDiscovering', true)},
+                '*:cancelDiscover': function() { this.set('isDiscovering', false)},
+                '*:contentDiscovered': function() { this.set('isDiscovering', false)},
+                '*:navigate': function () {
                     this.set('loading', true);
+                },
+
+            });
+
+            Y.on('hashchange', function (e) {
+                if ( app.get('isDiscovering') && e.newHash != app.get('hash') ) {
```

Comment:
If I understand correctly, this part is for closing the UDW when using the back button ? if so, this case should be handled separately in an improvement since it's an improvement not a bug. Let's focus only the navigation caused by the back button